backend/generator.js

```diff
@@ -35,14 +36,20 @@ exports.executeScript = function (socket, formData) {
 
   var process = spawn("./generate.sh", args);
 
-  process.stdout.on('data', function (data) {
-    console.log(data.toString());
-    socket.emit('logs', {donePercent: (donePercent = donePercent + 4), data: data.toString()});
+  readline.createInterface({
+    input     : process.stdout,
+    terminal  : false
+  }).on('line', function(data) {
+    console.log(data);
+    socket.emit('logs', {donePercent: (donePercent = donePercent + 4), data: data});
   });
 
-  process.stderr.on('data', function (data) {
-    console.log(data.toString());
-    socket.emit('err-logs', data.toString());
+  readline.createInterface({
```

Comment:
These are all very similar statements. would it be possible to instead have a single function and call it with appropriate params?